kong/runloop/balancer/init.lua

```diff
@@ -305,7 +305,9 @@ local function execute(balancer_data, ctx)
 
   if dns_cache_only then
     -- retry, so balancer is already set if there was one
-    balancer = balancer_data.balancer
+    balancer   = balancer_data.balancer
+    upstream   = balancer_data.upstream
+    hash_value = balancer_data.hash_value
```

Comment:
we may need to add a test case to cover: when use `consistent_hashing` to select peers in the balancer, `hash_value = balancer_data.hash_value` is required during the retry.

Reply:
I checked the code, it seems that even if leaving hash_value not be initialized here, it won't lead to a bug in the logic of consistent hashing.